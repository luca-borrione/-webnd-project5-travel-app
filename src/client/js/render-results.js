/* eslint-disable global-require, import/no-dynamic-require */

const renderThumbnail = ({
  city,
  flag,
  thumbnail = require('../assets/placeholder-destination-thumb.jpg'),
}) => `
<figure class="card__image">
  <img
    class="card__image-thumbnail"
    src="${thumbnail}"
    alt="view of the destination"
  />
  <figcaption class="card__image-caption">
    <span class="card__image-caption-flag">${flag}</span>
    <span class="card__image-caption-city">${city}</span>
  </figcaption>
</figure>
`;

const renderLocationInfo = ({
  continent,
  subregion,
  country,
  county,
  capital,
  currencies,
  languages,
  timezone,
  offset,
}) =>
  `<article class="card__info">${[
    { className: 'continent', label: 'Continent', value: continent },
    { className: 'subregion', label: 'Subregion', value: subregion },
    { className: 'country', label: 'Country', value: country },
    { className: 'county', label: 'County', value: county },
    { className: 'capital', label: 'Capital', value: capital },
    {
      className: 'currency',
      label: 'Currency',
      value: currencies.map(({ code, name }) => `${name} (${code})`).join(', '),
    },
    { className: 'language', label: 'Language', value: languages.join(', ') },
    { className: 'timezone-name', label: 'Timezone', value: timezone },
    { className: 'timezone-offset', label: 'Offset', value: offset },
  ]
    .map(
      ({ className, label, value }) =>
        value &&
        `
        <div class="card__info-${className}--label">${label}:</div>
        <div class="card__info-continent--value">${value}</div>
      `
    )
    .join('')}</article>`;

const getLocaleDateString = (date, timeZone) =>
  date.toLocaleDateString('en-GB', {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
    timeZone,
  });

const getLocaleTimeString = (date, timeZone) =>
  date.toLocaleTimeString('en-GB', {
    hour: '2-digit',
    hour12: true,
    minute: '2-digit',
    timeZone,
  });

const renderWeather = ({ className, dateFormatted, title, weather = '' }) => {
  const {
    // apparentMaxTemperature,
    // apparentMinTemperature,
    // apparentTemperature,
    description,
    humidity,
    icon,
    maxTemperature = '',
    minTemperature,
    temperature = '',
    windSpeed,
  } = weather;
  return `
  <article class="${className}">
    <main class="card__weather">
      <div class="card__weather-title">${title}</div>
      <div class="card__weather-date">${dateFormatted}</div>
      ${
        weather &&
        `<img
          class="card__weather-icon"
          src="${require(`../assets/weather-icons/${icon}.png`)}"
          alt="weather icon"
        />
        ${
          temperature &&
          `<div class="card__weather-temperature">
            <span>${temperature}<sup>&#8451;</sup></sup></span>
          </div>`
        }
        ${
          maxTemperature &&
          `<div class="card__weather-temperature">
            <span>max ${maxTemperature}<sup>&#8451;</sup></span>
            <span>min ${minTemperature}<sup>&#8451;</sup></span>
          </div>`
        }
        <div class="card__weather-humidity-wind">
          H ${Math.round(humidity)}% W ${Math.round(windSpeed)}&#x33A7;
        </div>
        <div class="card__weather-description">${description}</div>`
      }
    </main>
  </article>`;
};

const renderCurrentWeather = ({ dateString, timezone, ...currentWeather }) => {
  const currentDate = new Date(dateString);
  const currentDateFormatted = [
    getLocaleDateString(currentDate, timezone),
    getLocaleTimeString(currentDate, timezone),
  ].join(' - ');
  return renderWeather({
    className: 'current',
    title: 'Current Weather',
    dateFormatted: currentDateFormatted,
    weather: currentWeather,
  });
};

const renderDepartureInfo = ({ dateString, weather }) => {
  const departureDate = new Date(dateString);
  const timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
  const departureDateFormatted = [getLocaleDateString(departureDate, timezone)].join(' - ');
  return renderWeather({
    className: 'departure',
    title: 'Departure',
    dateFormatted: departureDateFormatted,
    weather,
  });
};

const renderReturnInfo = ({ dateString, weather }) => {
  const departureDate = new Date(dateString);
  const timezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
  const departureDateFormatted = [getLocaleDateString(departureDate, timezone)].join(' - ');
  return renderWeather({
    className: 'return',
    title: 'Return',
    dateFormatted: departureDateFormatted,
    weather,
  });
};

export const renderResultsView = ({
  capital,
  county,
  continent,
  country,
  city,
  currencies,
  languages,
  timezone,
  offset,
  flag,
  subregion,
  thumbnail,
  currentWeather,
  departureInfo,
  returnInfo,
}) => `
<main class="card">
  ${renderThumbnail({
    city,
    flag,
    thumbnail,
  })}
  ${renderLocationInfo({
    continent,
    subregion,
    country,
    county,
    capital,
    currencies,
    languages,
    timezone,
    offset,
  })}
  ${renderCurrentWeather(currentWeather)}
  ${renderDepartureInfo(departureInfo)}
  ${renderReturnInfo(returnInfo)}
</main>`;

/* eslint-enable global-require, import/no-dynamic-require */
