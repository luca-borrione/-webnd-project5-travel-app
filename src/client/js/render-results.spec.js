import { renderResultsView } from './render-results';

describe('renderResultsView', () => {
  const resultsViewParams = {
    thumbnail: 'mock-thumbnail-url',
    locationInfo: {
      capital: 'mock-capital',
      city: 'mock-city',
      continent: 'mock-continent',
      country: 'mock-country',
      county: 'mock-county',
      currencies: [
        {
          code: 'cur1',
          name: 'mock-currency1',
        },
        {
          code: 'cur2',
          name: 'mock-currency2',
        },
      ],
      flag: 'mock-flag',
      languages: ['mock-language1', 'mock-language2'],
      offset: 'mock-offset',
      subregion: 'mock-subregion',
      timezone: 'mock-timezone',
    },
    currentInfo: {
      timezone: 'Australia/Sydney',
      dateString: '2021-12-06 23:12',
      weather: {
        description: 'mock-description',
        humidity: 'mock-humidity',
        icon: 'a01d',
        temperature: 'mock-temperature',
        windSpeed: 'mock-wind-speed',
      },
    },
    departureInfo: {
      dateString: '2021-12-11',
      weather: {
        description: 'mock-description',
        humidity: 'mock-humidiy',
        icon: 'a01d',
        windSpeed: 'mock-wind-speed',
      },
    },
    returnInfo: {
      dateString: '2021-12-12',
      weather: {
        description: 'mock-description',
        humidity: 'mock-humidiy',
        icon: 'a01d',
        windSpeed: 'mock-wind-speed',
      },
    },
  };

  it.each`
    thumbnail
    ${'mock-thumbnail'}
    ${undefined}
  `('should correctly render the view, when the thumbnail is $thumbnail', ({ thumbnail }) => {
    const resultsView = renderResultsView({
      ...resultsViewParams,
      thumbnail,
    });
    expect(resultsView).toMatchSnapshot();
  });

  it.each`
    county
    ${'mock-county'}
    ${undefined}
  `('should correctly render the view, when the county is $county', ({ county }) => {
    const resultsView = renderResultsView({
      ...resultsViewParams,
      county,
    });
    expect(resultsView).toMatchSnapshot();
  });

  it('should correctly render the view, when the return weather forecast is undefined', () => {
    const resultsView = renderResultsView({
      ...resultsViewParams,
      returnInfo: {
        ...resultsViewParams.returnInfo,
        weather: undefined,
      },
    });
    expect(resultsView).toMatchSnapshot();
  });

  it('should correctly render the view, when both the departure and return weather forecasts are undefined', () => {
    const resultsView = renderResultsView({
      ...resultsViewParams,
      departureInfo: {
        ...resultsViewParams.departureInfo,
        weather: undefined,
      },
      returnInfo: {
        ...resultsViewParams.returnInfo,
        weather: undefined,
      },
    });
    expect(resultsView).toMatchSnapshot();
  });
});
