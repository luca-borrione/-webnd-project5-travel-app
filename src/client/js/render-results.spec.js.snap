// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderResultsView should correctly render the view, when both the departure and return weather forecasts are undefined 1`] = `
"
<main class=\\"card\\">
  
<figure class=\\"card__image\\">
  <img
    class=\\"card__image-thumbnail\\"
    src=\\"mock-thumbnail-url\\"
    alt=\\"view of the destination\\"
  />
  <figcaption class=\\"card__image-caption\\">
    <span class=\\"card__image-caption-flag\\">mock-flag</span>
    <span class=\\"card__image-caption-city\\">mock-city</span>
  </figcaption>
</figure>

  <article class=\\"card__info\\">
        <div class=\\"card__info-continent--label\\">Continent:</div>
        <div class=\\"card__info-continent--value\\">mock-continent</div>
      
        <div class=\\"card__info-subregion--label\\">Subregion:</div>
        <div class=\\"card__info-continent--value\\">mock-subregion</div>
      
        <div class=\\"card__info-country--label\\">Country:</div>
        <div class=\\"card__info-continent--value\\">mock-country</div>
      
        <div class=\\"card__info-county--label\\">County:</div>
        <div class=\\"card__info-continent--value\\">mock-county</div>
      
        <div class=\\"card__info-capital--label\\">Capital:</div>
        <div class=\\"card__info-continent--value\\">mock-capital</div>
      
        <div class=\\"card__info-currency--label\\">Currency:</div>
        <div class=\\"card__info-continent--value\\">mock-currency1 (cur1), mock-currency2 (cur2)</div>
      
        <div class=\\"card__info-language--label\\">Language:</div>
        <div class=\\"card__info-continent--value\\">mock-language1, mock-language2</div>
      
        <div class=\\"card__info-timezone-name--label\\">Timezone:</div>
        <div class=\\"card__info-continent--value\\">mock-timezone</div>
      
        <div class=\\"card__info-timezone-offset--label\\">Offset:</div>
        <div class=\\"card__info-continent--value\\">mock-offset</div>
      </article>
  
  <article class=\\"infobox current\\">
    <div class=\\"infobox__title\\">Current Weather</div>
    <div class=\\"infobox__date\\">7 Dec 2021 - 10:12 am</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    <div class=\\"weather__temperature\\">
      <span>mock-temperature<sup>&#8451;</sup></sup></span>
    </div>

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox departure\\">
    <div class=\\"infobox__title\\">Departure</div>
    <div class=\\"infobox__date\\">11 Dec 2021</div>
    
  </article>
  
  <article class=\\"infobox return\\">
    <div class=\\"infobox__title\\">Return</div>
    <div class=\\"infobox__date\\">12 Dec 2021</div>
    
  </article>
  
  <button class=\\"card__button\\" type=\\"button\\">
    <i class=\\"icon-heart-empty\\"></i>
    <i class=\\"icon-heart\\"></i>
    <span>Save Trip</span>
  </button>
  
</main>"
`;

exports[`renderResultsView should correctly render the view, when the county is mock-county 1`] = `
"
<main class=\\"card\\">
  
<figure class=\\"card__image\\">
  <img
    class=\\"card__image-thumbnail\\"
    src=\\"mock-thumbnail-url\\"
    alt=\\"view of the destination\\"
  />
  <figcaption class=\\"card__image-caption\\">
    <span class=\\"card__image-caption-flag\\">mock-flag</span>
    <span class=\\"card__image-caption-city\\">mock-city</span>
  </figcaption>
</figure>

  <article class=\\"card__info\\">
        <div class=\\"card__info-continent--label\\">Continent:</div>
        <div class=\\"card__info-continent--value\\">mock-continent</div>
      
        <div class=\\"card__info-subregion--label\\">Subregion:</div>
        <div class=\\"card__info-continent--value\\">mock-subregion</div>
      
        <div class=\\"card__info-country--label\\">Country:</div>
        <div class=\\"card__info-continent--value\\">mock-country</div>
      
        <div class=\\"card__info-county--label\\">County:</div>
        <div class=\\"card__info-continent--value\\">mock-county</div>
      
        <div class=\\"card__info-capital--label\\">Capital:</div>
        <div class=\\"card__info-continent--value\\">mock-capital</div>
      
        <div class=\\"card__info-currency--label\\">Currency:</div>
        <div class=\\"card__info-continent--value\\">mock-currency1 (cur1), mock-currency2 (cur2)</div>
      
        <div class=\\"card__info-language--label\\">Language:</div>
        <div class=\\"card__info-continent--value\\">mock-language1, mock-language2</div>
      
        <div class=\\"card__info-timezone-name--label\\">Timezone:</div>
        <div class=\\"card__info-continent--value\\">mock-timezone</div>
      
        <div class=\\"card__info-timezone-offset--label\\">Offset:</div>
        <div class=\\"card__info-continent--value\\">mock-offset</div>
      </article>
  
  <article class=\\"infobox current\\">
    <div class=\\"infobox__title\\">Current Weather</div>
    <div class=\\"infobox__date\\">7 Dec 2021 - 10:12 am</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    <div class=\\"weather__temperature\\">
      <span>mock-temperature<sup>&#8451;</sup></sup></span>
    </div>

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox departure\\">
    <div class=\\"infobox__title\\">Departure</div>
    <div class=\\"infobox__date\\">11 Dec 2021</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox return\\">
    <div class=\\"infobox__title\\">Return</div>
    <div class=\\"infobox__date\\">12 Dec 2021</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <button class=\\"card__button\\" type=\\"button\\">
    <i class=\\"icon-heart-empty\\"></i>
    <i class=\\"icon-heart\\"></i>
    <span>Save Trip</span>
  </button>
  
</main>"
`;

exports[`renderResultsView should correctly render the view, when the county is undefined 1`] = `
"
<main class=\\"card\\">
  
<figure class=\\"card__image\\">
  <img
    class=\\"card__image-thumbnail\\"
    src=\\"mock-thumbnail-url\\"
    alt=\\"view of the destination\\"
  />
  <figcaption class=\\"card__image-caption\\">
    <span class=\\"card__image-caption-flag\\">mock-flag</span>
    <span class=\\"card__image-caption-city\\">mock-city</span>
  </figcaption>
</figure>

  <article class=\\"card__info\\">
        <div class=\\"card__info-continent--label\\">Continent:</div>
        <div class=\\"card__info-continent--value\\">mock-continent</div>
      
        <div class=\\"card__info-subregion--label\\">Subregion:</div>
        <div class=\\"card__info-continent--value\\">mock-subregion</div>
      
        <div class=\\"card__info-country--label\\">Country:</div>
        <div class=\\"card__info-continent--value\\">mock-country</div>
      
        <div class=\\"card__info-county--label\\">County:</div>
        <div class=\\"card__info-continent--value\\">mock-county</div>
      
        <div class=\\"card__info-capital--label\\">Capital:</div>
        <div class=\\"card__info-continent--value\\">mock-capital</div>
      
        <div class=\\"card__info-currency--label\\">Currency:</div>
        <div class=\\"card__info-continent--value\\">mock-currency1 (cur1), mock-currency2 (cur2)</div>
      
        <div class=\\"card__info-language--label\\">Language:</div>
        <div class=\\"card__info-continent--value\\">mock-language1, mock-language2</div>
      
        <div class=\\"card__info-timezone-name--label\\">Timezone:</div>
        <div class=\\"card__info-continent--value\\">mock-timezone</div>
      
        <div class=\\"card__info-timezone-offset--label\\">Offset:</div>
        <div class=\\"card__info-continent--value\\">mock-offset</div>
      </article>
  
  <article class=\\"infobox current\\">
    <div class=\\"infobox__title\\">Current Weather</div>
    <div class=\\"infobox__date\\">7 Dec 2021 - 10:12 am</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    <div class=\\"weather__temperature\\">
      <span>mock-temperature<sup>&#8451;</sup></sup></span>
    </div>

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox departure\\">
    <div class=\\"infobox__title\\">Departure</div>
    <div class=\\"infobox__date\\">11 Dec 2021</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox return\\">
    <div class=\\"infobox__title\\">Return</div>
    <div class=\\"infobox__date\\">12 Dec 2021</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <button class=\\"card__button\\" type=\\"button\\">
    <i class=\\"icon-heart-empty\\"></i>
    <i class=\\"icon-heart\\"></i>
    <span>Save Trip</span>
  </button>
  
</main>"
`;

exports[`renderResultsView should correctly render the view, when the return weather forecast is undefined 1`] = `
"
<main class=\\"card\\">
  
<figure class=\\"card__image\\">
  <img
    class=\\"card__image-thumbnail\\"
    src=\\"mock-thumbnail-url\\"
    alt=\\"view of the destination\\"
  />
  <figcaption class=\\"card__image-caption\\">
    <span class=\\"card__image-caption-flag\\">mock-flag</span>
    <span class=\\"card__image-caption-city\\">mock-city</span>
  </figcaption>
</figure>

  <article class=\\"card__info\\">
        <div class=\\"card__info-continent--label\\">Continent:</div>
        <div class=\\"card__info-continent--value\\">mock-continent</div>
      
        <div class=\\"card__info-subregion--label\\">Subregion:</div>
        <div class=\\"card__info-continent--value\\">mock-subregion</div>
      
        <div class=\\"card__info-country--label\\">Country:</div>
        <div class=\\"card__info-continent--value\\">mock-country</div>
      
        <div class=\\"card__info-county--label\\">County:</div>
        <div class=\\"card__info-continent--value\\">mock-county</div>
      
        <div class=\\"card__info-capital--label\\">Capital:</div>
        <div class=\\"card__info-continent--value\\">mock-capital</div>
      
        <div class=\\"card__info-currency--label\\">Currency:</div>
        <div class=\\"card__info-continent--value\\">mock-currency1 (cur1), mock-currency2 (cur2)</div>
      
        <div class=\\"card__info-language--label\\">Language:</div>
        <div class=\\"card__info-continent--value\\">mock-language1, mock-language2</div>
      
        <div class=\\"card__info-timezone-name--label\\">Timezone:</div>
        <div class=\\"card__info-continent--value\\">mock-timezone</div>
      
        <div class=\\"card__info-timezone-offset--label\\">Offset:</div>
        <div class=\\"card__info-continent--value\\">mock-offset</div>
      </article>
  
  <article class=\\"infobox current\\">
    <div class=\\"infobox__title\\">Current Weather</div>
    <div class=\\"infobox__date\\">7 Dec 2021 - 10:12 am</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    <div class=\\"weather__temperature\\">
      <span>mock-temperature<sup>&#8451;</sup></sup></span>
    </div>

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox departure\\">
    <div class=\\"infobox__title\\">Departure</div>
    <div class=\\"infobox__date\\">11 Dec 2021</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox return\\">
    <div class=\\"infobox__title\\">Return</div>
    <div class=\\"infobox__date\\">12 Dec 2021</div>
    
  </article>
  
  <button class=\\"card__button\\" type=\\"button\\">
    <i class=\\"icon-heart-empty\\"></i>
    <i class=\\"icon-heart\\"></i>
    <span>Save Trip</span>
  </button>
  
</main>"
`;

exports[`renderResultsView should correctly render the view, when the thumbnail is mock-thumbnail 1`] = `
"
<main class=\\"card\\">
  
<figure class=\\"card__image\\">
  <img
    class=\\"card__image-thumbnail\\"
    src=\\"mock-thumbnail\\"
    alt=\\"view of the destination\\"
  />
  <figcaption class=\\"card__image-caption\\">
    <span class=\\"card__image-caption-flag\\">mock-flag</span>
    <span class=\\"card__image-caption-city\\">mock-city</span>
  </figcaption>
</figure>

  <article class=\\"card__info\\">
        <div class=\\"card__info-continent--label\\">Continent:</div>
        <div class=\\"card__info-continent--value\\">mock-continent</div>
      
        <div class=\\"card__info-subregion--label\\">Subregion:</div>
        <div class=\\"card__info-continent--value\\">mock-subregion</div>
      
        <div class=\\"card__info-country--label\\">Country:</div>
        <div class=\\"card__info-continent--value\\">mock-country</div>
      
        <div class=\\"card__info-county--label\\">County:</div>
        <div class=\\"card__info-continent--value\\">mock-county</div>
      
        <div class=\\"card__info-capital--label\\">Capital:</div>
        <div class=\\"card__info-continent--value\\">mock-capital</div>
      
        <div class=\\"card__info-currency--label\\">Currency:</div>
        <div class=\\"card__info-continent--value\\">mock-currency1 (cur1), mock-currency2 (cur2)</div>
      
        <div class=\\"card__info-language--label\\">Language:</div>
        <div class=\\"card__info-continent--value\\">mock-language1, mock-language2</div>
      
        <div class=\\"card__info-timezone-name--label\\">Timezone:</div>
        <div class=\\"card__info-continent--value\\">mock-timezone</div>
      
        <div class=\\"card__info-timezone-offset--label\\">Offset:</div>
        <div class=\\"card__info-continent--value\\">mock-offset</div>
      </article>
  
  <article class=\\"infobox current\\">
    <div class=\\"infobox__title\\">Current Weather</div>
    <div class=\\"infobox__date\\">7 Dec 2021 - 10:12 am</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    <div class=\\"weather__temperature\\">
      <span>mock-temperature<sup>&#8451;</sup></sup></span>
    </div>

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox departure\\">
    <div class=\\"infobox__title\\">Departure</div>
    <div class=\\"infobox__date\\">11 Dec 2021</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox return\\">
    <div class=\\"infobox__title\\">Return</div>
    <div class=\\"infobox__date\\">12 Dec 2021</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <button class=\\"card__button\\" type=\\"button\\">
    <i class=\\"icon-heart-empty\\"></i>
    <i class=\\"icon-heart\\"></i>
    <span>Save Trip</span>
  </button>
  
</main>"
`;

exports[`renderResultsView should correctly render the view, when the thumbnail is undefined 1`] = `
"
<main class=\\"card\\">
  
<figure class=\\"card__image\\">
  <img
    class=\\"card__image-thumbnail\\"
    src=\\"\\"
    alt=\\"view of the destination\\"
  />
  <figcaption class=\\"card__image-caption\\">
    <span class=\\"card__image-caption-flag\\">mock-flag</span>
    <span class=\\"card__image-caption-city\\">mock-city</span>
  </figcaption>
</figure>

  <article class=\\"card__info\\">
        <div class=\\"card__info-continent--label\\">Continent:</div>
        <div class=\\"card__info-continent--value\\">mock-continent</div>
      
        <div class=\\"card__info-subregion--label\\">Subregion:</div>
        <div class=\\"card__info-continent--value\\">mock-subregion</div>
      
        <div class=\\"card__info-country--label\\">Country:</div>
        <div class=\\"card__info-continent--value\\">mock-country</div>
      
        <div class=\\"card__info-county--label\\">County:</div>
        <div class=\\"card__info-continent--value\\">mock-county</div>
      
        <div class=\\"card__info-capital--label\\">Capital:</div>
        <div class=\\"card__info-continent--value\\">mock-capital</div>
      
        <div class=\\"card__info-currency--label\\">Currency:</div>
        <div class=\\"card__info-continent--value\\">mock-currency1 (cur1), mock-currency2 (cur2)</div>
      
        <div class=\\"card__info-language--label\\">Language:</div>
        <div class=\\"card__info-continent--value\\">mock-language1, mock-language2</div>
      
        <div class=\\"card__info-timezone-name--label\\">Timezone:</div>
        <div class=\\"card__info-continent--value\\">mock-timezone</div>
      
        <div class=\\"card__info-timezone-offset--label\\">Offset:</div>
        <div class=\\"card__info-continent--value\\">mock-offset</div>
      </article>
  
  <article class=\\"infobox current\\">
    <div class=\\"infobox__title\\">Current Weather</div>
    <div class=\\"infobox__date\\">7 Dec 2021 - 10:12 am</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    <div class=\\"weather__temperature\\">
      <span>mock-temperature<sup>&#8451;</sup></sup></span>
    </div>

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox departure\\">
    <div class=\\"infobox__title\\">Departure</div>
    <div class=\\"infobox__date\\">11 Dec 2021</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <article class=\\"infobox return\\">
    <div class=\\"infobox__title\\">Return</div>
    <div class=\\"infobox__date\\">12 Dec 2021</div>
    
  <aside class=\\"weather\\">
    <img
    class=\\"weather__icon\\"
    src=\\"\\"
    alt=\\"weather icon\\"
    />
    

    <div class=\\"weather__extra-info\\">
    H NaN% W NaN&#x33A7;
    </div>
    <div class=\\"weather__description\\">mock-description</div>
  </aside>
  </article>
  
  <button class=\\"card__button\\" type=\\"button\\">
    <i class=\\"icon-heart-empty\\"></i>
    <i class=\\"icon-heart\\"></i>
    <span>Save Trip</span>
  </button>
  
</main>"
`;
